File: 150. 逆波兰表达式求值/Solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i in "+-*/":
                if not stack:
                    return -1
                t1 = int(stack.pop())
                t2 = int(stack.pop())
                if i == '+':
                    t = t2 + t1
                elif i == '-':
                    t = t2 - t1
                elif i == '*':
                    t = t2 * t1
                else:
                    t = t2 / t1
                stack.append(t)
            else:
                stack.append(i)
        return int(stack.pop())

File: 150. 逆波兰表达式求值/Solution.cpp
#include<bits/stdc++.h>

using namespace std;


class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for(auto token: tokens){
            if(token == "+" || token == "-" || token == "*" || token == "/"){
                int t1 = st.top();
                st.pop();
                int t2 = st.top();  
                st.pop();
                int t = 0;
                if(token == "+") t = t2 + t1;
                else if(token == "-") t = t2 - t1;
                else if(token == "*") t = t2 * t1;
                else if(token == "/") t = t2 / t1;
                st.push(t);
            }
            else{
                st.push(stoi(token));
            }
        }
        return st.top();       
    }
};